File: docs/source/examples/CustomUserData/CreateCustomUserData.m
RLMApp *app = [RLMApp appWithId:@"<YourAppId>"]; // replace this with your App ID
[app loginWithCredential:[RLMCredentials anonymousCredentials] completion:^(RLMUser *user, NSError *error) {
    if (error != nil) {
        NSLog(@"Failed to log in: %@", error);
        return;
    }
    RLMMongoClient *client = [user mongoClientWithServiceName:@"mongodb-atlas"];
    RLMMongoDatabase *database = [client databaseWithName:@"my_database"];
    RLMMongoCollection *collection = [database collectionWithName:@"users"];
    [collection insertOneDocument:
        @{@"userId": [user identity], @"favoriteColor": @"pink"}
        completion:^(RLMObjectId *newObjectId, NSError *error) {
            if (error != nil) {
                NSLog(@"Failed to insert: %@", error);
            }
            NSLog(@"Inserted custom user data document with object ID: %@", newObjectId);
    }];
}];

File: source/examples/CustomUserData/CreateCustomUserData.swift
let appId = "<YourAppId>" // replace this with your App ID
let app = App(id: appId)
app.login(credentials: Credentials.anonymous()) { (user, error) in
    guard error == nil else {
        print("Failed to log in: \(error!.localizedDescription)")
        return
    }
    guard let user = user else {
        fatalError("User is nil without error")
    }
    let client = user.mongoClient("mongodb-atlas")
    let database = client.database(named: "my_database")
    let collection = database.collection(withName: "users")

    // Insert the custom user data object
    collection.insertOne([
        "userId": AnyBSON(user.identity!),
        "favoriteColor": "pink"
    ]) { (newObjectId, error) in
          guard error == nil else {
              print("Failed to insert document: \(error!.localizedDescription)")
              return
          }
          print("Inserted custom user data document with object ID: \(newObjectId!)")
    }
}
